pkg/operator/vspherecontroller/driver_starter.go

```diff
@@ -127,6 +130,34 @@ func (c *VSphereController) createCSIDriver() {
 	})
 }
 
+func (c *VSphereController) topologyHook(opSpec *operatorapi.OperatorSpec, deployment *appsv1.Deployment) error {
+	clusterCSIDriver, err := c.apiClients.ClusterCSIDriverInformer.Lister().Get(driverName)
+	if err != nil {
+		return err
+	}
+	vsphereConfig := clusterCSIDriver.Spec.DriverConfig
+	emptyDriverConfig := operatorapi.CSIDriverConfigSpec{}
+	if vsphereConfig == emptyDriverConfig {
+		return nil
+	}
+
+	if vsphereConfig.VSphere != nil {
+		topologyCategories := vsphereConfig.VSphere.TopologyCategories
+		if len(topologyCategories) > 0 {
+			containers := deployment.Spec.Template.Spec.Containers
+			for i := range containers {
+				if containers[i].Name != "csi-provisioner" {
+					continue
+				}
+
+				containers[i].Args = append(containers[i].Args, "--feature-gates=Topology=true", "--strict-topology")
```

Comment:
If I understand it correctly, the hook gets called in deployment controller and just modifies the manifest we provide if needed. So this hook can append `"--strict-topology"` flag to `"csi-provisioner"` container, but it will do so at runtime. 

So if the deployment manifest we provide is ever changed either by removing the flag, or uncommenting it, which would cause the flag to be present twice, it will break. Not sure we have that covered in CI.

Reply:
fair point. Although - I think most argument parsers will still parse these and set values only once. I *could* check for exact string being already in the arguments, but I am afraid this is also error prone. What happens if these flags get renamed?